lib/datadog/statsd/sender.rb

```diff
@@ -45,12 +45,18 @@ def start
 
       if CLOSEABLE_QUEUES
         def stop(join_worker: true)
+          message_queue = @message_queue
           message_queue.close if message_queue
+
+          sender_thread = @sender_thread
           sender_thread.join if sender_thread && join_worker
         end
       else
         def stop(join_worker: true)
+          message_queue = @message_queue
           message_queue << :close if message_queue
+
+          sender_thread = @sender_thread
           sender_thread.join if sender_thread && join_worker
         end
       end
```

Comment:
Minor: Both `stop` methods are quite similar and `CLOSEABLE_QUERIES` is a constant, so I suggest unifying them again and doing the `if CLOSEABLE_QUEUES` as a regular test inside the method itself.